C/services/south/ingest.cpp

```diff
@@ -1056,6 +1056,12 @@ void Ingest::useFilteredData(OUTPUT_HANDLE *outHandle,
 	lock_guard<mutex> guard(ingest->m_useDataMutex);
 	
 	vector<Reading *> *newData = readingSet->getAllReadingsPtr();
+	if (!ingest->m_data)
+	{
+		// If we are called during shutdown there will be no m_data in place
+		// and we create a new one to handle this special case.
+		ingest->m_data = new vector<Reading *>;
```

Comment:
If a new vector is allocated here, when is it deleted? I can see that this might be tricky because _m_data_ is sometimes allocated directly and at other times it's a copy of _m_queue_. If the _m_data_ allocated here is not deleted, it's a minor error because the plugin is shutting down anyway.

Reply:
It isn't, but this only occurs during a shutdown, so the process is about to exit and the memory will be cleaned then